Add python message generation

azeey opened the pull request:
# 🎉 New feature

## Summary
Adds python message generation using the old message generation framework. This is an alternative to #355 which uses the new framework.

## Test it
Run `ctest -R basic_TEST.py`.

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

azeey commented:
cc @Voldivh

Voldivh reviewed (approved):
LGTM! I tried it out locally and seems to be working.

Voldivh reviewed (commented):
Taking a closer look, I'm not sure why are we deleting everything related to Ruby and replacing it with Python instead of just adding the Python stuff.

azeey commented:
@Voldivh I've added options for where the python modules should be installed and added a test. Could you take another look?

> Taking a closer look, I'm not sure why are we deleting everything related to Ruby and replacing it with Python instead of just adding the Python stuff.

Support for Ruby was being removed in https://github.com/gazebosim/gz-msgs/pull/339, so I went ahead and removed it here as well.

Voldivh commented on python/CMakeLists.txt:
```diff
@@ -0,0 +1,34 @@
+# Append `_configured` to the file name so it doesn't interfere with tests
+set(python_init_file ${PROJECT_BINARY_DIR}/python/gz/${GS_DESIGNATION}/__init__.py_configured)
```
I'm not sure I'm understanding why it is needed to append the '_configured'? Why would it interfere with tests?

azeey replied:
Expanded the comment in https://github.com/gazebosim/gz-msgs/pull/362/commits/9b14d178c6673ef9b56c959b52868e54cc1199fe. Essentially the problem is bad interaction between python files in `build` and `install`

Voldivh reviewed (changes requested):
Just a couple of minor comments the rest LGTM!

azeey commented on python/CMakeLists.txt:
```diff
@@ -0,0 +1,34 @@
+  if(${PYTEST_result} EQUAL 0)
+    set(pytest_FOUND TRUE)
+  else()
+    message("")
```
Removed in https://github.com/gazebosim/gz-msgs/pull/362/commits/9b14d178c6673ef9b56c959b52868e54cc1199fe

Voldivh reviewed (approved):
LGTM!
